code/game/objects/items/tools/weldingtool.dm

```diff
@@ -102,6 +102,13 @@
 	qdel(src)
 
 /obj/item/weldingtool/attack(mob/living/carbon/human/H, mob/user)
+	if(isOn())
+		use(1)
+		var/turf/location = get_turf(user)
+		location.hotspot_expose(700, 50, 1)
+		if(get_fuel() <= 0)
+			set_light(0)
```

Comment:
Why the fucko is this even in afterattack to begin with? This should be called from use() or something?

Reply:
That's what I thought when I saw it